### PR TITLE
add Makefile

### PR DESCRIPTION
**Pull request by hellosmallstone**
Add a makefile, which includes building binary program, building docker, running unit-test and integration-test.
And add commit SHA and built time info in "tape version".

Signed-off-by: wuxu <wuxu1103@163.com>

### CODE REVIEW

File: pkg/infra/version.go [hunk 1]
```diff
@@ -7,17 +7,24 @@ import (
 
 const (
 	programName = "tape"
-	version     = "0.0.2"
+)
+
+var (
+	Version   string = "latest"
```
Comment (SamYuan1990) [inline pkg/infra/version.go:13]: version, built time, I am not sure for commitSHA.
I don't like update built time for each release. and make it had code here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# This makefile defines the following targets
+#   - tape - builds a binary program
+#   - tape-docker - build the tape image
+#   - unit-test - runs the go-test based unit tests
+#   - integration-test - runs the integration tests
+
+FABRIC_VERSION = latest
+INTERGATION_CASE = ANDLogic
+
+BASE_VERSION = 0.0.2
```
Comment (SamYuan1990) [inline Makefile:11]: so we need upgrade version in make file pre lease?
or will we use some git branch name? (get from CI?)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+GO_TAGS ?=
+
+export GO_LDFLAGS GO_TAGS FABRIC_VERSION INTERGATION_CASE
+
+.PHONY: tape
+tape:
+	@echo "Building tape program......"
+	go build -tags "$(GO_TAGS)" -ldflags "$(GO_LDFLAGS)" ./cmd/tape
+
+.PHONY: tape-docker
```
Comment (SamYuan1990) [inline Makefile:42]: make docker for short?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+tape-docker:
+	@echo "Building tape docker......"
+	docker build .
+
+.PHONY: unit-test
+unit-test:
```
Comment (SamYuan1990) [inline Makefile:48]: or `make unittest`? is it possible?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+GO_LDFLAGS = $(patsubst %,-X $(PROJECT_NAME)/pkg/infra.%,$(METADATA_VAR))
+GO_LDFLAGS += -X '$(PROJECT_NAME)/pkg/infra.BuiltTime=$(BuiltTime)'
+
+GO_TAGS ?=
+
+export GO_LDFLAGS GO_TAGS FABRIC_VERSION INTERGATION_CASE
+
+.PHONY: tape
+tape:
+	@echo "Building tape program......"
```
Comment (SamYuan1990) [inline Makefile:39]: question, can we use `make tape` without given `GO_TAGS` and `GO_LDFLAGS`?

Comment (hellosmallstone) [reply Makefile:39]: Currently GO_TAGS is empty, which can be deleted. 
GO_LDFLAGS assigns Version, CommitSHA and BuiltTime in program.

**Review by guoger — APPROVED** [review]

**Comment by guoger** [comment]
LGTM, it would be nice to have a `make install` to cp binary to local `GOBIN` :)

**Pull request merged**